draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1551,76 +1555,92 @@ user agent might wish to block sending cookies during "third-party" requests
 from setting cookies (see {{third-party-cookies}}).
 
 If the user agent does attach a Cookie header field to an HTTP request, the
-user agent MUST send the cookie-string (defined below) as the value of the
-header field.
+user agent MUST send the cookie-string as the value of the
+header field. The cookie-string is computed as defined in {{retrieval-algorithm}},
+whereby the retrieval-uri is defined as the request-uri and the same-site status
+is computed for the HTTP request as defined in {{same-site-requests}}.
 
-The user agent MUST use an algorithm equivalent to the following algorithm to
-compute the cookie-string from a cookie store and a request-uri:
+### Non-HTTP APIs {#non-http}
+
+The user agent may implement "non-HTTP" APIs that can be used to access
+stored cookies.
 
-1.  Let cookie-list be the set of cookies from the cookie store that meets all
-    of the following requirements:
+A user agent MAY return an empty cookie-string in certain contexts, such as when
+retrieval occurs within a third-party context (see {{third-party-cookies}}).
 
-    *   Either:
+If a user agent does return cookies for a given call to a "non-HTTP" API with an
+associated Document, then the user agent MUST compute the cookie-string following
+the algorithm defined in {{retrieval-algorithm}, whereby the retrieval-uri is the
+associated Document's URL. The retrieval is same-site if the Document's "site for
+cookies" is same-site with the top-level origin as defined in {{document-requests}}.
 
-        *   The cookie's host-only-flag is true and the canonicalized
-            request-host is identical to the cookie's domain.
+### Retrieval Algorithm {#retrieval-algorithm}
 
-        Or:
+The user agent MUST use an algorithm equivalent to the following algorithm to
+compute the cookie-string from a cookie store, a retrieval-uri, and the retrieval's
+same-site status.
```

Comment:
Also, nit, that you can ignore since this language is used elsewhere in the spec, but the "use an algorithm equivalent to this algorithm" reads a little awkward. Perhaps some variation like the following is a little bit better?

"Given a cookie store, a retrieval-uri, and the retrieval's same-site status, the following algorithm returns a cookie-string" (or something)